app/src/main/java/fr/free/nrw/commons/coordinates/CoordinateEditHelper.java

```diff
@@ -0,0 +1,109 @@
+package fr.free.nrw.commons.coordinates;
+
+import static fr.free.nrw.commons.notification.NotificationHelper.NOTIFICATION_EDIT_COORDINATES;
+
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+import fr.free.nrw.commons.BuildConfig;
+import fr.free.nrw.commons.Media;
+import fr.free.nrw.commons.R;
+import fr.free.nrw.commons.actions.PageEditClient;
+import fr.free.nrw.commons.notification.NotificationHelper;
+import fr.free.nrw.commons.utils.ViewUtilWrapper;
+import io.reactivex.Observable;
+import io.reactivex.Single;
+import java.util.Objects;
+import javax.inject.Inject;
+import javax.inject.Named;
+import timber.log.Timber;
+
+public class CoordinateEditHelper {
+    private final NotificationHelper notificationHelper;
+    public final PageEditClient pageEditClient;
+    private final ViewUtilWrapper viewUtil;
+
+    @Inject
+    public CoordinateEditHelper(final NotificationHelper notificationHelper,
+        @Named("commons-page-edit") final PageEditClient pageEditClient,
+        final ViewUtilWrapper viewUtil) {
+        this.notificationHelper = notificationHelper;
+        this.pageEditClient = pageEditClient;
+        this.viewUtil = viewUtil;
+    }
+
+    /**
+     * Public interface to edit coordinates
+     * @param context to be added
+     * @param media to be added
+     * @param Accuracy to be added
+     * @return Single<Boolean>
+     */
+    public Single<Boolean> makeCoordinatesEdit(final Context context, final Media media,
+        final String Latitude, final String Longitude, final String Accuracy) {
+        viewUtil.showShortToast(context,
+            context.getString(R.string.coordinates_edit_helper_make_edit_toast));
+        return addCoordinates(media, Latitude, Longitude, Accuracy)
+            .flatMapSingle(result -> Single.just(showCoordinatesEditNotification(context, media,
+                result)))
+            .firstOrError();
+    }
+
+    /**
+     * Replaces new coordinates
+     * @param media to be added
+     * @param Latitude to be added
+     * @param Longitude to be added
+     * @param Accuracy to be added
+     * @return Observable<Boolean>
+     */
+    private Observable<Boolean> addCoordinates(final Media media, final String Latitude,
+        final String Longitude, final String Accuracy) {
+        Timber.d("thread is coordinates adding %s", Thread.currentThread().getName());
+        final String summary = "Adding Coordinates";
+
+        final StringBuilder buffer = new StringBuilder();
```

Comment:
Please initialise the buffer with the current wikitext, then search/replace location inside it.
It is OK to post the whole thing even though only location changed, that's what already happens when you edit via the website interface.

Reply:
I believe we already have a utility to get the current wikitext of an image, we use it to display the information in the Media Details activity.